### ISSUE TITLE
video tutorial for server deployment

### ISSUE THREAD

**Issue by veonhoon** [DeviceFarmer/stf#594]
Hi i was able to use stf local without any problems but when i want to deploy it onto server, I'm having issues. 

the dev that I'm working with said that i needed a static IP if i wanted to access from other network. 

So i even purchased a static IP, but he's still not able to get it to work. 

is there a video tutorial that i can follow or something? thanks in advance, also if someone can help me, i will compensate you for your time. thank you in advance.

**Comment by jsjs4013** [comment]
Hi

stf local --public-ip <your_internal_network_ip_here>

Does not it work?

Thanks

**Comment by veonhoon** [comment]
> Hi @veonhoon
> 
> stf local --public-ip <your_internal_network_ip_here>
> 
> Does not it work?
> 
> Thanks

yeah I've tried this, but even if i can access the device from WAN, the device screen is grey and doesn't show up.

**Comment by jachstet-sea** [comment]
Do you have some kind of NAT or firewall in between? Would you need some kind of port forwarding?

**Comment by denis99999** [comment]
@veonhoon , you need to configure the machine (i.e. router or server) that carries your static IP address to allow the following incoming ports (by default):
- 7100
- 7110
- 7400-7700

and in case your STF machine is behind a router you have to forward (NAT) these ports to the STF Machine

**Comment by veonhoon** [comment]
Thank you very much. I've followed the comments and i'm able to access the dashboard and the devices through the internet. 

But i noticed that the phones are very slow, compared to controlling them from LAN. Is this normal?

**Comment by denis99999** [comment]
@veonhoon ,
It depends on the network, its bandwidth and the end-to-end latency between the client and the STF server, you can improve this aspect in two ways, on the one hand by playing on the `--screen-jpeg-quality` option, and on the other hand maybe trying another access network like a 4G or 5G network, but in this case you would need the corresponding router with a static IPv4 address (I mean without CGNAT) or with an IPv6 address (but I've never tested that with STF), it's tempting!
